# HG changeset patch
# User Francois Bissey <francois.bissey@canterbury.ac.nz>
# Date 1305856872 -43200
# Node ID 4a494081f153b93bb973b3ce97d5f8463055a152
# Parent  ee672d4d5c0c98a4869635c88636f806cd52420e
#11377 scrubbing and putting some order in module_list.py. Not strictly necessary but nice.

diff --git a/module_list.py b/module_list.py
--- a/module_list.py
+++ b/module_list.py
@@ -14,9 +14,9 @@
     sys.exit(1)
 else:
     SAGE_ROOT  = os.environ['SAGE_ROOT']
-    SAGE_LOCAL = SAGE_ROOT + '/local/'
-    SAGE_DEVEL = SAGE_ROOT + '/devel/'
-    SAGE_INC = SAGE_LOCAL + 'include/'
+    SAGE_LOCAL = SAGE_ROOT + '/local'
+    SAGE_DEVEL = SAGE_ROOT + '/devel'
+    SAGE_INC = SAGE_LOCAL + '/include/'
 
 
 #########################################################
@@ -45,33 +45,6 @@
 
 
 #########################################################
-### Debian-related stuff
-#########################################################
-
-if os.environ.has_key('SAGE_DEBIAN'):
-    debian_include_dirs=["/usr/include",
-                         "/usr/include/cudd",
-                         "/usr/include/eclib",
-                         "/usr/include/FLINT",
-                         "/usr/include/fplll",
-                         "/usr/include/givaro",
-                         "/usr/include/gmp++",
-                         "/usr/include/gsl",
-                         "/usr/include/linbox",
-                         "/usr/include/NTL",
-                         "/usr/include/numpy",
-                         "/usr/include/pari",
-                         "/usr/include/polybori",
-                         "/usr/include/polybori/groebner",
-                         "/usr/include/singular",
-                         "/usr/include/singular/singular",
-                         "/usr/include/symmetrica",
-                         "/usr/include/zn_poly"]
-    include_dirs = include_dirs + debian_include_dirs
-else:
-    debian_include_dirs=[]
-
-#########################################################
 ### Commonly used definitions
 #########################################################
 
@@ -80,9 +53,9 @@
 # timestamp of the numpy build.
 numpy_depends = [SAGE_LOCAL + '/lib/python/site-packages/numpy/core/include/numpy/_numpyconfig.h']
 
-flint_depends = [SAGE_LOCAL + '/include/FLINT/flint.h']
-singular_depends = [SAGE_LOCAL + '/include/libsingular.h']
-ginac_depends = [SAGE_LOCAL + '/include/pynac/ginac.h']
+flint_depends = [SAGE_INC + 'FLINT/flint.h']
+singular_depends = [SAGE_INC + 'libsingular.h']
+ginac_depends = [SAGE_INC + 'pynac/ginac.h']
 
 #########################################################
 ### PolyBoRi defines
@@ -124,7 +97,7 @@
                sources = ['sage/algebras/quatalg/quaternion_algebra_element.pyx'],
                language='c++',
                libraries = ["csage", "flint", "gmp", "gmpxx", "m", "stdc++", "ntl"],
-               include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+               include_dirs = [SAGE_INC + 'FLINT/'],
                depends = flint_depends),
 
     Extension('sage.algebras.quatalg.quaternion_algebra_cython',
@@ -340,9 +313,9 @@
     Extension('sage.graphs.matchpoly',
               sources = ['sage/graphs/matchpoly.pyx'],
               libraries = ['gmp', 'flint'],
-              include_dirs = [SAGE_ROOT + '/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               extra_compile_args = ['-std=c99'],
-              depends = [SAGE_ROOT + "/local/include/FLINT/flint.h"]),
+              depends = flint_depends),
 
     Extension('sage.graphs.planarity',
               sources = ['sage/graphs/planarity/graphColorVertices.c',
@@ -527,20 +500,20 @@
     Extension('sage.libs.ecl',
               sources = ["sage/libs/ecl.pyx"],
               libraries = ["ecl"],
-              include_dirs = [SAGE_ROOT+'/local/include/ecl/'],
-              depends = [SAGE_ROOT + '/local/include/ecl/ecl.h']),
+              include_dirs = [SAGE_INC + 'ecl/'],
+              depends = [SAGE_INC + 'ecl/ecl.h']),
 
     Extension('sage.libs.flint.flint',
               sources = ["sage/libs/flint/flint.pyx"],
               libraries = ["csage", "flint", "gmp", "gmpxx", "m", "stdc++"],
-              include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               extra_compile_args=["-std=c99", "-D_XPG6"],
               depends = flint_depends),
 
     Extension('sage.libs.flint.fmpz_poly',
               sources = ["sage/libs/flint/fmpz_poly.pyx"],
               libraries = ["csage", "flint", "gmp", "gmpxx", "m", "stdc++"],
-              include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               extra_compile_args=["-std=c99", "-D_XPG6"],
               depends = flint_depends),
     
@@ -548,8 +521,8 @@
               sources = ['sage/libs/fplll/fplll.pyx'],
               libraries = ['gmp', 'mpfr', 'stdc++', 'fplll'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/fplll'],
-              depends = [SAGE_ROOT + "/local/include/fplll/fplll.h"]),
+              include_dirs = [SAGE_INC + 'fplll'],
+              depends = [SAGE_INC + "fplll/fplll.h"]),
 
     Extension('sage.libs.linbox.linbox',
               sources = ['sage/libs/linbox/linbox.pyx'],
@@ -564,7 +537,7 @@
               sources = ['sage/libs/lcalc/lcalc_Lfunction.pyx'],
               libraries = ['m', 'ntl', 'mpfr', 'gmp', 'gmpxx',
                            'Lfunction', 'stdc++'],
-              include_dirs = [SAGE_ROOT + "/local/include/lcalc/"],
+              include_dirs = [SAGE_INC + "lcalc/"],
               extra_compile_args=["-O3", "-ffast-math"],
               language = 'c++'),
      
@@ -573,7 +546,7 @@
     Extension('sage.libs.libecm',
               sources = ['sage/libs/libecm.pyx'],
               libraries = ['ecm', 'gmp'],
-              depends = [SAGE_ROOT + "/local/include/ecm.h"]),
+              depends = [SAGE_INC + "ecm.h"]),
      
     Extension('sage.libs.mwrank.mwrank',
               sources = ["sage/libs/mwrank/mwrank.pyx",
@@ -598,60 +571,60 @@
               sources = ['sage/libs/ppl.pyx', 'sage/libs/ppl_shim.cc'],
               libraries = ['ppl', 'gmpxx', 'gmp', 'm'], 
               language="c++",
-              depends = [SAGE_LOCAL + "/include/ppl.hh"]),
+              depends = [SAGE_INC + "ppl.hh"]),
 
     Extension('sage.libs.ratpoints',
               sources = ["sage/libs/ratpoints.pyx"],
-              depends = [SAGE_ROOT + '/local/include/ratpoints.h'],
+              depends = [SAGE_INC + 'ratpoints.h'],
               libraries = ["ratpoints", "gmp"]),
     
     Extension('sage.libs.singular.singular',
               sources = ['sage/libs/singular/singular.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'ntl', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends),
     
     Extension('sage.libs.singular.polynomial',
               sources = ['sage/libs/singular/polynomial.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends),
 
     Extension('sage.libs.singular.ring',
               sources = ['sage/libs/singular/ring.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends),
 
     Extension('sage.libs.singular.groebner_strategy',
               sources = ['sage/libs/singular/groebner_strategy.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends),
 
     Extension('sage.libs.singular.function',
               sources = ['sage/libs/singular/function.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC +'singular'],
               depends = singular_depends),
 
     Extension('sage.libs.singular.option',
               sources = ['sage/libs/singular/option.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends),
 
     Extension('sage.libs.symmetrica.symmetrica',
               sources = ["sage/libs/symmetrica/%s"%s for s in ["symmetrica.pyx"]],
               include_dirs = ['/usr/include/malloc/'],
               libraries = ["symmetrica"],
-              depends = [SAGE_ROOT + "/local/include/symmetrica/def.h"]),
+              depends = [SAGE_INC + "symmetrica/def.h"]),
 
     Extension('sage.libs.mpmath.utils',
               sources = ["sage/libs/mpmath/utils.pyx"],
@@ -898,7 +871,7 @@
     Extension('sage.matrix.matrix_mod2_dense',
               sources = ['sage/matrix/matrix_mod2_dense.pyx'],
               libraries = ['gmp','m4ri', 'gd', 'png12', 'z'],
-              depends = [SAGE_ROOT + "/local/include/png.h", SAGE_ROOT + "/local/include/m4ri/m4ri.h"]),
+              depends = [SAGE_INC + "png.h", SAGE_INC + "m4ri/m4ri.h"]),
 
     Extension('sage.matrix.matrix_modn_dense',
               sources = ['sage/matrix/matrix_modn_dense.pyx'],
@@ -912,7 +885,7 @@
               sources = ['sage/matrix/matrix_mpolynomial_dense.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends),
 
     #Extension('sage.matrix.matrix_pid_dense',
@@ -1039,14 +1012,14 @@
     Extension('sage.modular.modform.eis_series_cython',
               sources = ['sage/modular/modform/eis_series_cython.pyx'],
               libraries = ["gmp", "flint"],
-              include_dirs = [SAGE_ROOT + '/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               extra_compile_args = ['-std=c99'],
               depends = flint_depends),
 
     Extension('sage.modular.modsym.apply',
               sources = ['sage/modular/modsym/apply.pyx'],
               libraries = ["csage", "flint", "gmp", "gmpxx", "m", "stdc++"],
-              include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               extra_compile_args=["-std=c99",  "-D_XPG6"],
               depends = flint_depends),
 
@@ -1056,7 +1029,7 @@
     Extension('sage.modular.modsym.heilbronn',
               sources = ['sage/modular/modsym/heilbronn.pyx'],
               libraries = ["csage", "flint", "gmp", "gmpxx", "m", "stdc++"],
-              include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               extra_compile_args=["-std=c99", "-D_XPG6"],
               depends = flint_depends),
 
@@ -1099,7 +1072,7 @@
     Extension('sage.modules.vector_mod2_dense',
               sources = ['sage/modules/vector_mod2_dense.pyx'],
               libraries = ['gmp','m4ri', 'png12', 'gd'],
-              depends = [SAGE_ROOT + "/local/include/png.h", SAGE_ROOT + "/local/include/m4ri/m4ri.h"]),
+              depends = [SAGE_INC + "png.h", SAGE_INC + "m4ri/m4ri.h"]),
     
     Extension('sage.modules.vector_rational_dense',
               sources = ['sage/modules/vector_rational_dense.pyx'],
@@ -1124,17 +1097,17 @@
 
     Extension("sage.numerical.mip",
               ["sage/numerical/mip.pyx"],
-            include_dirs=["local/include/"],
-            libraries=["csage","stdc++"]),
+              include_dirs=[SAGE_INC],
+              libraries=["csage","stdc++"]),
 
     Extension("sage.numerical.backends.generic_backend",
               ["sage/numerical/backends/generic_backend.pyx"],
-              include_dirs = [SAGE_ROOT+"/local/include/", "sage/c_lib/include/"],
+              include_dirs = [SAGE_INC, "sage/c_lib/include/"],
               libraries=["csage", "stdc++"]),
 
     Extension("sage.numerical.backends.glpk_backend",
               ["sage/numerical/backends/glpk_backend.pyx"],
-              include_dirs = [SAGE_ROOT+"/local/include/", "sage/c_lib/include/"],
+              include_dirs = [SAGE_INC, "sage/c_lib/include/"],
               language = 'c++',
               libraries=["csage", "stdc++", "glpk", "gmp", "z"]),
 
@@ -1247,7 +1220,7 @@
               sources = ['sage/rings/fraction_field_FpT.pyx'],
               libraries = ["csage", "flint", "gmp", "gmpxx", "ntl", "zn_poly"],
               extra_compile_args=["-std=c99", "-D_XPG6"],
-              include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               depends = flint_depends),
     
     Extension('sage.rings.laurent_series_ring_element',
@@ -1449,14 +1422,14 @@
               sources = ['sage/rings/polynomial/multi_polynomial_ideal_libsingular.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends),
 
     Extension('sage.rings.polynomial.multi_polynomial_libsingular',
               sources = ['sage/rings/polynomial/multi_polynomial_libsingular.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
+              include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends),
 
     Extension('sage.rings.polynomial.multi_polynomial_ring_generic',
@@ -1488,14 +1461,14 @@
               sources = ['sage/rings/polynomial/polynomial_zmod_flint.pyx'],
               libraries = ["csage", "flint", "gmp", "gmpxx", "ntl", "zn_poly"],
               extra_compile_args=["-std=c99", "-D_XPG6"],
-              include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               depends = flint_depends),
 
     Extension('sage.rings.polynomial.polynomial_integer_dense_flint',
               sources = ['sage/rings/polynomial/polynomial_integer_dense_flint.pyx'],
               language = 'c++',
               libraries = ["csage", "flint", "ntl", "gmpxx", "gmp"],
-              include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+              include_dirs = [SAGE_INC + 'FLINT/'],
               depends = flint_depends),
 
     Extension('sage.rings.polynomial.polynomial_integer_dense_ntl',
@@ -1509,7 +1482,7 @@
               language = 'c++',
               extra_compile_args=["-std=c99"] + uname_specific('SunOS', [], ['-D_XPG6']),
               libraries = ["csage", "flint", "ntl", "gmpxx", "gmp"],
-              include_dirs = [SAGE_ROOT + '/local/include/FLINT/', SAGE_ROOT + '/devel/sage/sage/libs/flint/'],
+              include_dirs = [SAGE_INC + 'FLINT/', 'sage/libs/flint/'],
               depends = flint_depends),
 
     Extension('sage.rings.polynomial.polynomial_modn_dense_ntl',
@@ -1521,11 +1494,11 @@
     Extension('sage.rings.polynomial.pbori',
               sources = ['sage/rings/polynomial/pbori.pyx'],
               libraries=(['polybori','pboriCudd', 'groebner', 'gd', 'png12', 'm4ri']), 
-              include_dirs = [SAGE_ROOT+'/local/include/cudd',
-                              SAGE_ROOT+'/local/include/polybori',
-                              SAGE_ROOT+'/local/include/polybori/groebner',
+              include_dirs = [SAGE_INC + 'cudd',
+                              SAGE_INC + 'polybori',
+                              SAGE_INC + 'polybori/groebner',
                               "sage/libs/polybori"],
-              depends = [SAGE_ROOT + "/local/include/polybori/polybori.h"],
+              depends = [SAGE_INC + "polybori/polybori.h"],
               extra_compile_args = polybori_extra_compile_args,
               language = 'c++'),
 
@@ -1551,10 +1524,10 @@
     Extension('sage.schemes.elliptic_curves.descent_two_isogeny',
               sources = ['sage/schemes/elliptic_curves/descent_two_isogeny.pyx'],
               extra_compile_args=["-std=c99"],
-              depends = [SAGE_ROOT + '/local/include/ratpoints.h',
-                         SAGE_ROOT + '/local/include/gmp.h',
-                         SAGE_ROOT + '/local/include/FLINT/flint.h'],
-              include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
+              depends = [SAGE_INC + 'ratpoints.h',
+                         SAGE_INC + 'gmp.h'] +
+                         flint_depends,
+              include_dirs = [SAGE_INC + 'FLINT/'],
               libraries = ['flint', 'gmp', 'ratpoints']),
 
     Extension('sage.schemes.generic.toric_divisor_class',
@@ -1714,12 +1687,12 @@
 from sage.misc.package import is_package_installed
 
 
-if (os.path.isfile(SAGE_ROOT+"/local/include/cplex.h") and                                                                                                                                                                                   
-    os.path.isfile(SAGE_ROOT+"/local/lib/libcplex.a")):
+if (os.path.isfile(SAGE_INC + "cplex.h") and                                                                                                                                                                                   
+    os.path.isfile(SAGE_LOCAL + "/lib/libcplex.a")):
     ext_modules.append(
         Extension("sage.numerical.backends.cplex_backend",                                                                                                                                                                                    
                   ["sage/numerical/backends/cplex_backend.pyx"],                                                                                                                                                                              
-                  include_dirs = [SAGE_ROOT+"/local/include/","/sage/c_lib/include/"],                                                                                                                                                        
+                  include_dirs = [SAGE_INC, "sage/c_lib/include/"],                                                                                                                                                        
                   language = 'c',                                                                                                                                                                                                             
                   libraries = ["csage", "stdc++", "cplex"]) 
         )
@@ -1728,7 +1701,7 @@
     ext_modules.append(
         Extension("sage.numerical.backends.coin_backend",                                                                                                                                                                                     
                   ["sage/numerical/backends/coin_backend.pyx"],
-                  include_dirs = [SAGE_ROOT+"/local/include/","sage/c_lib/include/"],
+                  include_dirs = [SAGE_INC, "sage/c_lib/include/"],
                   language = 'c++',
                   libraries = ["csage", "stdc++", "Cbc", "CbcSolver", "Cgl", "Clp", "CoinUtils", "OsiCbc", "OsiClp", "Osi", "OsiVol", "Vol"])
         )
